### DISCUSSION TITLE
Running sshd from cmd.exe running as system doesn't work

### DISCUSSION THREAD

**Discussion by LubosD** [PowerShell/Win32-OpenSSH#2059] [General]
Hi,

I'm trying to debug some configuration issues with regards to integrating sshd into a larger remote control package. For that I tried to run sshd.exe directly from cmd.exe running as `nt authority\system` (spawned via `psexec`) assuming it should operate just fine. It doesn't.

I tried both sshd.exe bundled with Win10 and a recent release from this repository - both behave the same way.

Logging in to sshd executed this way fails:
```
Server listening on 0.0.0.0 port 22.
debug3: fd 5 is not O_NONBLOCK
debug1: Server will not fork when running in debugging mode.
debug3: send_rexec_state: entering fd = 8 config len 329
debug3: ssh_msg_send: type 0
debug3: send_rexec_state: done
Connection from fd00:1::1ac0:4dff:fe60:fc9e port 39294 on fd00:1::5054:ff:fe56:db1f port 22
debug1: Local version string SSH-2.0-OpenSSH_for_Windows_8.1
debug1: Remote protocol version 2.0, remote software version OpenSSH_9.3
debug1: match: OpenSSH_9.3 pat OpenSSH* compat 0x04000000
debug2: fd 5 setting O_NONBLOCK
debug3: spawning "C:\\Windows\\system32\\OpenSSH\\sshd.exe" -dddd -D -y
debug2: Network child is on pid 10368
debug3: send_rexec_state: entering fd = 4 config len 329
debug3: ssh_msg_send: type 0
debug3: write ERROR from cb(2):232, io:000001BB74636580
ssh_msg_send: write
send_rexec_state: ssh_msg_send failed
debug1: do_cleanup
debug1: Killing privsep child 10368
```

The SSH client sees a connection reset:
```
debug1: kex_input_ext_info: publickey-hostbound@openssh.com=<0>
debug3: receive packet: type 6
debug2: service_accept: ssh-userauth
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug3: send packet: type 50
Connection reset by fd00:1::5054:ff:fe56:db1f port 22
```

However, when the very same executable is run as a service as Local System, it works just fine.

What's the difference? What can I do to make sshd work when run from the command line directly?

**Comment by LubosD** [comment]
I'll answer myself: the problem seems to be the `-i` flag passed to psexec.

**Comment by tgauth** [comment]
Sounds like you figured it out - https://github.com/PowerShell/Win32-OpenSSH/wiki/Troubleshooting-Steps also details running sshd as SYSTEM via psexec
